solr/solrj/src/test/org/apache/solr/client/solrj/impl/ClusterStateProviderTest.java

```diff
@@ -117,4 +149,39 @@ private Instant getCreationTimeFromClusterStatus(String collectionName)
     Map<String, Object> collection = (Map<String, Object>) collections.get(collectionName);
     return Instant.ofEpochMilli((long) collection.get("creationTimeMillis"));
   }
+
+  @Test
+  public void testClusterStateProvider() throws SolrServerException, IOException {
```

Comment:
In the process of writing this test, I discovered some keys like "health" in the base ZkNodeProps that prevented equivalency with the ZK one.  So I enhanced HttpCSP to remove these keys.